### DISCUSSION TITLE
Demo code for callback function Python -> C++ -> Python

### DISCUSSION THREAD

**Discussion by jwuttke** [RosettaCommons/binder#189] [General]
To prepare for full-scale adoption of binder/pybind11 in a large project of ours,
I wrote a short sandbox example that demonstrates how a Python-defined
callback function can be used with a C++-defined class constructor,
https://jugit.fz-juelich.de/mlz/sandbox/binder-sandbox/-/tree/main/callback.

Maybe you are interested to add this to the collection of binder examples.

The ease of this solution is a big advantage of binder/pybind11 over Swig.

**Comment by lyskov** [comment]
Sure, more examples would be great and PR adding this will be welcome! A few thoughts: current implementation (https://jugit.fz-juelich.de/mlz/sandbox/binder-sandbox/-/tree/main/callback) have ~9+ small files while total amount of code is minimal which makes reading it a bit difficult. So perhaps converting this whole set to a documentation page (with embedded code) might make a bit more sense and will be more readable. Also, i personally find use of word `callback` here a bit confusing since this pattern is a bit more general, perhaps "binding and using `std::function` types" will be a better title?
